report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+			$L \in Lambda$		  & Lambda expression\\
+			$e \in Exp$			  & Expression\\
+			$el \in Lit$		  & Expression Literal\\
+			$p \in Pnames$        & Procedure names\\
+			$L \in Lambda$        & Statement Lambdas\\
+			$S \in Stm$           & Statement\\
+			$f \in Fnames$ 		  & Function names\\
+			$D_V \in Dcl_V$		  & Variable declarations\\
+			$D_P \in Dcl_P$       & Function declarations\\
+			$T \in Type$          & Types
+		\end{tabular}
+	\end{table}
+\end{center}
+
+For the rest of this section, it will be assumed that functions exists with domains equal to the sets of literals associated with the various expressions' syntax categories, and with co-domains equal to the values extracted from those literals.
+The abstract syntax of HCL can be seen below.
+
+$a\ ::=\ n\ |\ x\ |\ a_1+a_2\ |\ a_1-a_2\ |\ a_1*a_2\ |\ a_1/a_2\ |\ (a)$\\
+$b\ ::=\ tf\ |\ x\ |\ a_1<a_2\ |\ a_1>a_2\ |\ a_1<=a_2\ |\ a_1>=a_2\ |\ a_1==a_2\ |\ b_1\ and\ b_2\ |\ b_1\ or\ b_2\ |\ b\ not\ |\ (b)$\\
+$t\ ::=\ str\ |\ x\ |\ t_1+t_2\ |\ t\ toString\ |\ (t)$\\
+$list\ ::=\ lst\ |\ x\ |\ list\ length\ |\ list\ @\ a\ |\ (list)$\\
+$tuple\ ::=\ tpl\ |\ x\ |\ tuple\ @\ a\ |\ (tuple)$\\
+$e\ ::=\ a\ |\ b\ |\ t\ |\ list\ |\ tuple\ |\ F_{exp}\ | L$\\
+$pOrf\ ::=\ f\ |\ p$\\
+$eOrL\ ::=\ e\ |\ L$\\
+$F_{exp}\ ::=\ f\ |\ eOrL\ f\ eOrL^*$\\
+$L\ ::= \{\ S\ \}\ |\ \{\ S\ return\ e\ \}\ |\ \{\ S;\ e\ \}$\\
+$S\ ::=\ x\ =\ e\ |\ S_1;S_2\ |\ b\ then\ L\ |\ L\ while\ b\ |\ p\ |\ eOrL\ p\ eOrL^*\ |\ D_V\ |\ D_P\ |\ \epsilon$\\
+$D_v\ ::=\ T\ x\ =\ e;\ S\ |\ T\ x\ = call_{exp};\ S\ |\ T\ x;\ D_v\ |\ \epsilon$\\
+$D_p\ ::=\ T\ pOrf\ =\ ([T\ x]\sim):\ T\ L;\ S\ |\ T\ pOrf;\ S\ |\ pOrf\ =\ ([T\ x]\sim):\ T\ L;\ S\ |\ \epsilon$
+
+\textbf{\large{Conventions}} \\
+For each variable $x$, an arbitrary variable-environment $env_V$ describes which address $adr$, $x$ is bound to.
+For simplicity, the set of all possible addresses, denoted $Adr$, is assumed to be equal to the set of all integers.
+\begin{center}
+	$Adr = \mathbb{Z}$
+\end{center}
+The set of all variable-environments, denoted \textbf{$Env_V$}, is the set of all partial functions from variables to addresses.
+\begin{center}
+	$Env_V = Var \cup \{next\} \rightharpoonup Adr$
+\end{center}
+For every $adr$, the function $new : Adr \rightarrow Adr$ returns the next $adr$, whether it is free or not.
+\begin{center}
+	$new (adr) = adr + 1$
+\end{center}
+A store $sto$ describes what values $v$ are pointed to by the addresses found in $Adr$.
+
+With the variable bindings found in $env_V$ and the address content found in $sto$, the expressions are capable of returning the values of the $sto$.
+This means that every transition of expressions must be on the form.
+\begin{center}
+	$env_V,sto \vdash e \rightarrow v$
+\end{center} 
+
+\subsection{Big-Step Semantic for Expressions}
+An expression is any syntactic component that evaluates to a value.
+In HCL most expressions behave the same, but a few transition-systems uses special notation.
```

Comment:
i might be stupid but what do you mean with "In HCL most expressions behave the same" - doesn't all expression behave the same? being the whole point of our language?

Reply:
Half right. By that sentence I mean that they are denoted the same in the rules, but I can see how it makes no sense when read.
Fixed